packages/react-native-web/src/modules/usePlatformMethods/index.js

```diff
@@ -43,22 +44,33 @@ function setNativeProps(node, nativeProps, classList, pointerEvents, style, prev
  * Adds non-standard methods to the hode element. This is temporarily until an
  * API like `ReactNative.measure(hostRef, callback)` is added to React Native.
  */
-export default function usePlatformMethods(props: Object) {
-  const previousStyleRef = useRef(null);
-  const { classList, style, pointerEvents } = props;
+export default function usePlatformMethods({
+  classList,
+  pointerEvents,
+  style
+}: {
+  classList?: Array<string | boolean>,
+  style?: $PropertyType<ViewProps, 'style'>,
+  pointerEvents: $PropertyType<ViewProps, 'pointerEvents'>
+}) {
+  const previousStyleRef = React.useRef(null);
+  const setNativePropsArgsRef = React.useRef(null);
+  setNativePropsArgsRef.current = { classList, pointerEvents, style };
 
-  return useMemo(
-    () => (hostNode: any) => {
+  return React.useCallback(
```

Comment:
We should use React.useRef here instead to create this function once and return the .current of that ref. This will guarantee ref equality but always use the latest values. since this matches the intent and correctly is guaranteed by the behavior of useRef I think it’s better than relying on memoization which is not guaranteed ( https://reactjs.org/docs/hooks-reference.html  )

useCallback in particular seems odd here semantically too.

Reply:
Got it. Updated!